bayesian/utility/v2/sa_response_builder.py

```diff
@@ -92,6 +101,33 @@ def _raise_if_inprogress_or_timeout(self):
                 logger.warning(error_message)
                 raise SARBRequestInprogressException(error_message)
 
+    def _get_analysed_dependencies(self, analysed_dependencies):
+        """Filter report fields based on user status."""
+        if g.user_status != UserStatus.REGISTERED:
+            freetier_user_fields = ['cve_ids', 'cvss', 'cwes',
+                                    'cvss_v3', 'severity', 'title', 'id', 'url']
+            for dependency in analysed_dependencies:
+                self._filter_fields(freetier_user_fields, dependency)
+
+        return analysed_dependencies
+
+    def _filter_fields(self, required_fields, dependency):
+        """Keep only freetier fields for given dependency."""
+        for index, public_vul in enumerate(dependency.get('public_vulnerabilities', [])):
```

Comment:
This loop can be written once with the public and private vulnerabilities parameterized.